### PR TITLE
[17.0][MIG] delivery_state

### PR DESCRIPTION
**Pull request by pablo-marin-s73**
Standard migration delivery_state

### CODE REVIEW

**Review by hildickethan — APPROVED** [review]
code + functional

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The procedure hasn't been followed and the "pre-commit auto-fixes" commit is missing.

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

**Comment by sbiosca-s73** [comment]
@pedrobaeza changes done

File: delivery_state/data/mail_template.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <odoo noupdate="1">
     <record id="delivery_state_delivered_notification" model="mail.template">
         <field name="name">Delivery: Picking delivered by Email</field>
-        <field name="model_id" ref="delivery.model_stock_picking" />
+        <field name="model_id" ref="stock_delivery.model_stock_picking" />
```
Comment (pedrobaeza) [inline delivery_state/data/mail_template.xml]: I think we should put the base one:
```suggestion
        <field name="model_id" ref="stock.model_stock_picking" />
```

File: delivery_state/views/res_config_settings_view.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
         <field name="model">res.config.settings</field>
         <field name="inherit_id" ref="stock.res_config_settings_view_form" />
         <field name="arch" type="xml">
-            <xpath expr="//div[@id='stock_move_email']" position="after">
+            <xpath expr="//setting[@id='stock_move_email']" position="after">
                 <div class="col-12 col-lg-6 o_setting_box" id="delivery:state_email">
```
Comment (pedrobaeza) [inline delivery_state/views/res_config_settings_view.xml]: From the migration guide:

imagen

File: delivery_state/views/res_config_settings_view.xml [hunk 3]
```diff
@@ -25,7 +25,7 @@
                         </div>
                         <div
                             class="row mt16"
-                            attrs="{'invisible': [('delivery_state_delivered_email_validation', '=', False)]}"
+                            invisible="delivery_state_delivered_email_validation == False"
```
Comment (pedrobaeza) [inline delivery_state/views/res_config_settings_view.xml]:
```suggestion
                            invisible="not delivery_state_delivered_email_validation"
```

File: delivery_state/views/res_config_settings_view.xml [hunk 4]
```diff
@@ -35,7 +35,7 @@
                             <field
                                 name="delivery_state_delivered_mail_template_id"
                                 class="oe_inline"
-                                attrs="{'required': [('delivery_state_delivered_email_validation', '=', True)]}"
+                                required="delivery_state_delivered_email_validation == True"
```
Comment (pedrobaeza) [inline delivery_state/views/res_config_settings_view.xml]:
```suggestion
                                required="delivery_state_delivered_email_validation"
```

And the same for the rest.

**Comment by sbiosca-s73** [comment]
@pedrobaeza done

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**